### DISCUSSION TITLE
Pipeline creation times improvement with Naga on vange-rs

### DISCUSSION THREAD

**Discussion by kvark** [gfx-rs/wgpu#1431] [Show and tell]
I'm working on `profiling` integration, and I got some first numbers of performance comparison of our shader pipeline on a real-ish project: https://github.com/kvark/vange-rs. This is just about creating 2 shader modules (from SPIR-V!) and a render pipeline. Very primitive and artificial test, take the numbers with a grain of salt!

SPIRV-Cross: **5.1ms**
<img width="1086" alt="time-cross" src="https://user-images.githubusercontent.com/107301/116275397-98a3e700-a751-11eb-9d03-b4d72fa771bd.png">

Naga: **3.0ms**
<img width="1147" alt="time-naga" src="https://user-images.githubusercontent.com/107301/116275438-a194b880-a751-11eb-8658-48ca873e874d.png">

Common part (calling Metal routines): **1.55ms**

That leaves us with a rough upper bound on the time spent in Naga and SPIRV-Cross respectively being 1.45ms and 3.55ms. This tells us Naga is **2.5x** faster, roughly, at least with regards to how it's used in `wgpu` tech stack.

**Comment by simast** [comment]
Keep up the amazing work you are doing around WebGPU! This library is a gem for Rust community.
